### ISSUE TITLE
panicking while loading www.lemonde.fr

### ISSUE THREAD

**Issue by lhirlimann** [servo/servo#10949]
```
[ludo@Oulanl servo]$ ./mach run --release http://www.lemonde.fr
ERROR:js::rust: Error at http://s1.lemde.fr/bootstrap/list-early-load.js:0:0: document.write is not a function

ERROR:js::rust: Error at http://www.lemonde.fr/:5:6: d.write is not a function

ERROR:js::rust: Error at http://www.lemonde.fr/:0:0: lmd.conf is undefined

ERROR:js::rust: Error at http://www.lemonde.fr/:6:0: lmd.conf is undefined

ERROR:js::rust: Error at http://s1.lemde.fr/medias/web/1.2.691/js/lmd/core/ux/sync.js:323:15: global.lmd.conf is undefined

ERROR:js::rust: Error at http://www.lemonde.fr/:1:0: lmd.conf is undefined

ERROR:js::rust: Error at http://www.lemonde.fr/:0:0: lmd.conf is undefined

ERROR:js::rust: Error at http://www.lemonde.fr/:0:0: lmd.conf is undefined

ERROR:js::rust: Error at http://www.lemonde.fr/:0:0: lmd.conf is undefined

ERROR:js::rust: Error at http://s1.lemde.fr/package/www/libs-5f5ac021d8e536cee9ad8326e66e2e0f.js:38:194: conf is null

ERROR:js::rust: Error at http://s1.lemde.fr/min/medias/web/f4071b6bcae6621d0d012bb77abf2611/js/lmd/core/metrics/ligatus.js:0:34: conf is null

ERROR:js::rust: Error at http://s1.lemde.fr/min/medias/web/0fa82ca6a24f0226d2ac36e1cbc4425c/js/lmd/core/metrics/aisle.js:0:34: conf is null

ERROR:js::rust: Error at http://s1.lemde.fr/min/medias/web/f44632262c411eb7b118dff29c3aa712/js/lmd/core/metrics/ignition-one.js:0:159: lmd.conf is undefined

ERROR:js::rust: Error at http://www.lemonde.fr/:53:24: lmd.conf is undefined

thread 'ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }' panicked at 'assertion failed: *self.stack == mem::transmute(&*self)', /home/ludo/servo/.cargo/git/checkouts/rust-mozjs-ebb4917e843c0a11/master/src/rust.rs:376
stack backtrace:
ERROR:compositing::constellation: Panic: DOMRefCell<T> already mutably borrowed
ERROR:compositing::constellation: Backtrace:
frame #0  - 0x000055e5f85fb18d - backtrace::backtrace::trace::h8cb7b0465fa42e9f
frame #1  - 0x000055e5f85fb115 - backtrace::capture::Backtrace::new::ha2e43e88f310ebee
frame #2  - 0x000055e5f77aa2ab - _<F as std..boxed..FnBox<A>>::call_box::hade68e206dc3d945
frame #3  - 0x000055e5f7f6eb6c - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h230b30f93b5b89ae
frame #4  - 0x000055e5f861c963 - std::sys_common::unwind::begin_unwind_inner::h30e12d15ce2b2e25
frame #5  - 0x000055e5f861e3a8 - std::sys_common::unwind::begin_unwind_fmt::hb2de8a9968d38523
frame #6  - 0x000055e5f862e511 - rust_begin_unwind
frame #7  - 0x000055e5f866e25f - core::panicking::panic_fmt::h257ceb0aa351d801
frame #8  - 0x000055e5f8679e84 - core::option::expect_failed::h2d57a5644f345e0b
frame #9  - 0x000055e5f7603044 - _<dom..bindings..cell..DOMRefCell<T>>::borrow::he99b96d1d212b2ae
frame #10 - 0x000055e5f76030a0 - script::dom::htmlanchorelement::HTMLAnchorElement::update_href::hc27282abcb83f88c
frame #11 - 0x000055e5f75870d0 - _<dom..htmlanchorelement..HTMLAnchorElement as dom..bindings..codegen..Bindings..HTMLAnchorElementBinding..HTMLAnchorElementMethods>::SetHash::h73dc6e717728f70c
frame #12 - 0x000055e5f7586fe4 - script::dom::bindings::codegen::Bindings::HTMLAnchorElementBinding::set_hash::h8b8545fdedc0784d
frame #13 - 0x000055e5f7fc62f5 - CallJitSetterOp
frame #14 - 0x000055e5f7685328 - script::dom::bindings::utils::call_setter::hba2ed818ad9709da
frame #15 - 0x000055e5f7685204 - script::dom::bindings::utils::generic_call::h7189873cda5d34a8
frame #16 - 0x000055e5f80d3651 - 2js12CallJSNativeEP9JSContextPFbS1_jPN2JS5ValueEERKNS2_8CallArgs
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/jscntxtinlines.h:235
                               - 2js6InvokeEP9JSContextN2JS8CallArgsENS_14MaybeConstruct
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:502
frame #17 - 0x000055e5f80d4a6d - 2js6InvokeEP9JSContextRKN2JS5ValueES5_jPS4_NS2_13MutableHandleIS3_E
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:558
frame #18 - 0x000055e5f80d7c7d - 2js20InvokeGetterOrSetterEP9JSContextP8JSObjectN2JS5ValueEjPS5_NS4_13MutableHandleIS5_E
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:628
frame #19 - 0x000055e5f8128966 - 2js5Shape3setEP9JSContextN2JS6HandleIPNS_12NativeObjectEEENS4_IP8JSObjectEENS3_13MutableHandleINS3_5ValueEEERNS3_14ObjectOpResult
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Shape-inl.h:51
frame #20 - 0x000055e5f811ac72 - SetExistingProperty
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/NativeObject.cpp:2194
                               - 2js17NativeSetPropertyEP9JSContextN2JS6HandleIPNS_12NativeObjectEEENS3_IP8JSObjectEENS3_I4jsidEENS_13QualifiedBoolENS2_13MutableHandleINS2_5ValueEEERNS2_14ObjectOpResult
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/NativeObject.cpp:2230
frame #21 - 0x000055e5f8096789 - SetObjectProperty
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:324
frame #22 - 0x000055e5f80c7811 - SetPropertyOperation
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:361
                               - Interpret
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:2492
frame #23 - 0x000055e5f80d334c - 2js9RunScriptEP9JSContextRNS_8RunState
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:452
frame #24 - 0x000055e5f80d35b7 - 2js6InvokeEP9JSContextN2JS8CallArgsENS_14MaybeConstruct
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:521
frame #25 - 0x000055e5f80d4a6d - 2js6InvokeEP9JSContextRKN2JS5ValueES5_jPS4_NS2_13MutableHandleIS3_E
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/vm/Interpreter.cpp:558
frame #26 - 0x000055e5f83ac075 - _Z20JS_CallFunctionValueP9JSContextN2JS6HandleIP8JSObjectEENS2_INS1_5ValueEEERKNS1_16HandleValueArrayENS1_13MutableHandleIS6_EE
                               @ /home/ludo/servo/.cargo/git/checkouts/mozjs-06d7f04b6dbb8a8e/master/mozjs/js/src/jsapi.cpp:4336
frame #27 - 0x000055e5f757f716 - script::dom::bindings::codegen::Bindings::EventHandlerBinding::EventHandlerNonNull::Call::h218486f4e5d032a3
frame #28 - 0x000055e5f74db5ed - script::dom::eventdispatcher::invoke::hc758defe9d93e5cf
frame #29 - 0x000055e5f74dc760 - script::dom::eventdispatcher::dispatch_event::hee15b1cfae16c60a
frame #30 - 0x000055e5f775ca12 - script::dom::htmlscriptelement::HTMLScriptElement::dispatch_event::h4958093d6cad890b
frame #31 - 0x000055e5f7734cff - script::dom::htmlscriptelement::HTMLScriptElement::execute::he5548d7006646b04
frame #32 - 0x000055e5f7732e31 - script::dom::document::Document::process_asap_scripts::h81b6bf19e7345d54
frame #33 - 0x000055e5f77320e1 - script::dom::document::Document::finish_load::hb896be6e7638c4e0
frame #34 - 0x000055e5f7758cf1 - _<dom..htmlscriptelement..ScriptContext as net_traits..AsyncResponseListener>::response_complete::h24cc8de0a0e7bef7
frame #35 - 0x000055e5f7ab6fc9 - net_traits::ResponseAction::process::h96c9a519767b3f60
frame #36 - 0x000055e5f775c557 - _<network_listener..ListenerRunnable<T> as script_thread..Runnable>::handler::hef5bbcfe0e5588da
frame #37 - 0x000055e5f77b5778 - script::script_thread::ScriptThread::handle_msg_from_script::hdb2373cca373c061
frame #38 - 0x000055e5f78057eb - script::script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::h0047e195408ce412
frame #39 - 0x000055e5f77f2d11 - script::script_thread::ScriptThread::handle_msgs::h64e3903ea6a9416f
frame #40 - 0x000055e5f77a8687 - std::sys_common::unwind::try::try_fn::h13d13124daf8b86a
frame #41 - 0x000055e5f862e49b - __rust_try
frame #42 - 0x000055e5f862e42d - std::sys_common::unwind::inner_try::h47a4d9cd4a369dcd
frame #43 - 0x000055e5f77a9a9f - _<F as std..boxed..FnBox<A>>::call_box::h8bc3d601084a39ef
frame #44 - 0x000055e5f8634dc4 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
frame #45 - 0x00007f423fe85589 - start_thread
frame #46 - 0x00007f423f9a45cc - __clone
frame #47 - 0x0000000000000000 - <unknown>

   1:     0x55e5f862f2c0 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
   2:     0x55e5f86368db - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
   3:     0x55e5f8636543 - std::panicking::default_hook::hc2c969e7453d080c
   4:     0x55e5f7f6eba6 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h230b30f93b5b89ae
   5:     0x55e5f861c963 - std::sys_common::unwind::begin_unwind_inner::h30e12d15ce2b2e25
   6:     0x55e5f6dcdb8f - std::sys_common::unwind::begin_unwind::h65390c63b501d6ed
   7:     0x55e5f7671734 - js..jsapi..Rooted<*mut js..jsapi..JSObject>::drop.116673::h8368e7e705b3b73c
   8:     0x55e5f74dbd01 - script::dom::eventdispatcher::invoke::hc758defe9d93e5cf
   9:     0x55e5f74dc760 - script::dom::eventdispatcher::dispatch_event::hee15b1cfae16c60a
  10:     0x55e5f775ca12 - script::dom::htmlscriptelement::HTMLScriptElement::dispatch_event::h4958093d6cad890b
  11:     0x55e5f7734cff - script::dom::htmlscriptelement::HTMLScriptElement::execute::he5548d7006646b04
  12:     0x55e5f7732e31 - script::dom::document::Document::process_asap_scripts::h81b6bf19e7345d54
  13:     0x55e5f77320e1 - script::dom::document::Document::finish_load::hb896be6e7638c4e0
  14:     0x55e5f7758cf1 - _<dom..htmlscriptelement..ScriptContext as net_traits..AsyncResponseListener>::response_complete::h24cc8de0a0e7bef7
  15:     0x55e5f7ab6fc9 - net_traits::ResponseAction::process::h96c9a519767b3f60
  16:     0x55e5f775c557 - _<network_listener..ListenerRunnable<T> as script_thread..Runnable>::handler::hef5bbcfe0e5588da
  17:     0x55e5f77b5778 - script::script_thread::ScriptThread::handle_msg_from_script::hdb2373cca373c061
  18:     0x55e5f78057eb - script::script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::h0047e195408ce412
  19:     0x55e5f77f2d11 - script::script_thread::ScriptThread::handle_msgs::h64e3903ea6a9416f
  20:     0x55e5f77a8687 - std::sys_common::unwind::try::try_fn::h13d13124daf8b86a
  21:     0x55e5f862e49b - __rust_try
  22:     0x55e5f862e42d - std::sys_common::unwind::inner_try::h47a4d9cd4a369dcd
  23:     0x55e5f77a9a9f - _<F as std..boxed..FnBox<A>>::call_box::h8bc3d601084a39ef
  24:     0x55e5f8634dc4 - std::sys::thread::Thread::new::thread_start::h6f266e069bf4ec2b
  25:     0x7f423fe85589 - start_thread
  26:     0x7f423f9a45cc - __clone
  27:                0x0 - <unknown>
thread panicked while panicking. aborting.
Servo exited with return value -4
[ludo@Oulanl servo]$ 
```

**Comment by jdm** [comment]
I stopped at rust_panic in a debugger and looked at a backtrace; this should be fixed by #10903. Thanks for reporting it!

**Comment by jdm** [comment]
Fixed by #10903.
